pubtools/_quay/manifest_list_merger.py

```diff
@@ -0,0 +1,114 @@
+from copy import deepcopy
+import logging
+
+from .quay_client import QuayClient
+
+LOG = logging.getLogger()
+logging.basicConfig()
+LOG.setLevel(logging.INFO)
+
+
+class ManifestListMerger:
+    """Class containing logic for merging manifest lists of two images."""
+
+    def __init__(
+        self, src_image, dest_image, quay_username=None, quay_password=None, host=None
+    ):
+        """
+        Initialize.
+
+        Args:
+            src_image (str):
+                Address to a new image whose manifest list contains the newer data.
+            dest_image (str):
+                Address to an older image whose data will be overwritten.
+            quay_username (str):
+                Username to login to quay. If ommited, external client instance should be set.
+            quay_password (str):
+                Password to login to quay. If ommited, external client instance should be set.
+            host (str):
+                Custom hostname to connect to. If ommited, standard quay.io will be used.
+        """
+        self.src_image = src_image
+        self.dest_image = dest_image
+        if quay_username and quay_password:
+            self._quay_client = QuayClient(quay_username, quay_password, host)
+        else:
+            self._quay_client = None
+
+    def set_quay_client(self, quay_client):
+        """
+        Set client instance to be used for the HTTP API operations.
+
+        Args:
+            quay_client (QuayClient):
+                Instance of QuayClient.
+        """
+        self._quay_client = quay_client
+
+    def merge_manifest_lists(self):
+        """Merge manifest lists and upload to Quay. Main entrypoint method."""
+        if not self._quay_client:
+            raise RuntimeError("QuayClient instance must be set")
+
+        LOG.info(
+            "Merging manifest lists of images '{0}' and '{1}'".format(
+                self.src_image, self.dest_image
+            )
+        )
+        src_manifest_list = self._quay_client.get_manifest_list(self.src_image)
+        dest_manifest_list = self._quay_client.get_manifest_list(self.dest_image)
```

Comment:
Does it need to support dest image not having mainfest list?

Reply:
Based on the discussions with Dynamo folks, the answer is 'no'.